File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/bridge/uielement/Input.java
package com.bigboxcode.designpattern.bridge.uielement;

public class Input extends UIElement{

    public Input(ColorSchema colorSchema) {
        super(colorSchema);
    }

    @Override
    public void printElement() {
        // Write code for printing element
        System.out.println("Printing Input");

        // Set color schema
        colorSchema.setColor();
    }
}

File: design-patterns/ts/bridge/ui-element/input.ts
// input.ts

import Color from "./color";
import UIElement from "./ui-element";


class Input extends UIElement {
    constructor(color: Color) {
        super(color);
    }
    public printElement(): void {
        this.color.setColor();

        console.log("Printing Input");
    }
}

export default Input;